tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -174,6 +175,116 @@ DECL_CONVERT_OP(FakeQuant);
 
 #undef DECL_CONVERT_OP
 
+LogicalResult ConvertTFLGeluOp::matchAndRewrite(
+    Operation* op, PatternRewriter& rewriter) const {
+  auto tfl_gelu_op = cast<TFL::GeluOp>(op);
+  Location loc = op->getLoc();
+
+  Value input = tfl_gelu_op.input();
+  ShapedType input_type = input.getType().dyn_cast<ShapedType>();
+  ShapedType output_type =
+      tfl_gelu_op.getResult().getType().dyn_cast<ShapedType>();
+  // Not a ranked tensor output
```

Comment:
ShapedType doesn't check for rank, so this would assert on line 190 if unranked tensor was given as input.

Reply:
It is embarrassing. Cast to RankedTensorType instead of ShapedType.